torch/csrc/jit/script/init.cpp

```diff
@@ -51,26 +52,43 @@ struct VISIBILITY_HIDDEN PythonValue : public SugaredValue {
   PythonValue(py::object self)
   : self(std::move(self)) {}
 
-  std::pair<std::vector<TypePtr>, TypePtr> getFunctionType(size_t n_args, size_t n_binders) {
+  std::tuple<std::vector<TypePtr>, std::vector<TypePtr>> getArgumentTypes(size_t n_args, size_t n_binders) {
```

Comment:
Hmm my intention when I put a single return type in here is that having multiple outputs in Python (and those are our reference semantics) is really equivalent to returning a tuple of those elements (while taking multiple arguments is not fundamentally equivalent to taking a single tuple).

Reply:
We have the following constraints:

1. we are going to check FunctionSchema objects to report error messages. In fact the current PythonValue::call checking code is highly redundant with tryEmitSchema, and is less functional: it is missing a bunch of conversion logic that must run. 
2. FunctionSchema represent elements of Graphs which natively have multiple outputs.
3. When we are binding a python function into a Graph we are generating a wrapper around it so that it works as a PythonOp in the Graph.

In graphs, FunctionSchema, and PythonOp, things can return multiple values. The odd one out here is python functions. When we natively support tuples, PythonOp should always return a _single argument_ that might be a IR Tuple. However, it is important that we transition this over to use FunctionSchema for the reasons laid out above. So in the meantime it makes sense to flatten the tuples so we can continue to call PythonOps until we patch in native support for Tuples.